### PR TITLE
refactor: :recycle: refactor some utils

### PR DESCRIPTION
**Pull request by HannesOberreiter**
- converted `carts` logic into a class `ShoppingCart`
- simplified `currentDate()` & `getQueryParameter()` functions (note: `new URL` browser support, see <https://caniuse.com/?search=URL>)

### CODE REVIEW

File: site/cart.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ async function load() {
     let cart = null;
     const cartName = getQueryParameter("name");
     if (cartName) {
-        for (c of carts) {
+        for (c of shoppingCart.carts) {
```
Comment (badlogic) [inline site/cart.js:11]: Please rename `shoppingCart` to `shoppingCarts` as the instance manages multiple carts.

Similarly, please also rename the class `ShoppingCart` to `ShoppingCarts`

File: site/utils.js [hunk 2]
```diff
@@ -38,100 +38,136 @@ const stores = {
...
+  }
+  return items;
 }
 
-let carts = [];
-loadCarts();
-
-function loadCarts() {
+/**
+ * @description Class for managing the shopping cart, which is stored in local storage
```
Comment (badlogic) [inline site/utils.js:112]: It manages multiple carts, please use the plural form for the class name and docs.

File: site/utils.js [hunk 3]
```diff
@@ -38,100 +38,136 @@ const stores = {
...
-        item.dateOldest = item.priceHistory[item.priceHistory.length - 1].date;
-        item.date = item.priceHistory[0].date;
-        let highestPriceBefore = -1;
-        for (let i = 1; i < item.priceHistory.length; i++) {
-            const price = item.priceHistory[i];
-            highestPriceBefore = Math.max(highestPriceBefore, price.price);
-        }
-        if (highestPriceBefore == -1) highestPriceBefore = item.price;
-        item.highestBefore = highestPriceBefore;
+  const response = await fetch("latest-canonical.json");
```
Comment (badlogic) [inline site/utils.js:89]: Please use the same indentation as found in the rest of the file(s), i.e. 4 spaces.

**Review by badlogic — CHANGES_REQUESTED** [review]
Looking good, just a few minor change requests.

File: site/utils.js (hunk 3 above)
Comment (HannesOberreiter) [reply site/utils.js]: Sorry habit, was my editor. Changed back and added an config file https://github.com/badlogic/heissepreise/pull/39/commits/84ee1577cc34037a9796af3c84c44a6ee7dd2947

File: site/cart.js (hunk 1 above)
Comment (HannesOberreiter) [reply site/cart.js]: Changed class name and variable instance.

File: site/utils.js (hunk 2 above)
Comment (HannesOberreiter) [reply site/utils.js]: changed `jsdoc`

**Comment by badlogic** [comment]
Cheers!

**Pull request merged**